security/pkg/nodeagent/secretfetcher/secretfetcher.go

```diff
@@ -325,25 +351,46 @@ func (sf *SecretFetcher) scrtUpdated(oldObj, newObj interface{}) {
 			Version:      t.String(),
 		}
 		sf.secrets.Store(rootCertResourceName, *nsRoot)
-		log.Debugf("secret %s is updated", rootCertResourceName)
+		log.Infof("secret %s is updated", rootCertResourceName)
 		if sf.UpdateCache != nil {
 			sf.UpdateCache(rootCertResourceName, *nsRoot)
 		}
 	}
 }
 
-// FindIngressGatewaySecret returns the secret for a k8sKeyA, or empty secret if no
+// FindIngressGatewaySecret returns the secret whose name matches the key, or empty secret if no
 // secret is present. The ok result indicates whether secret was found.
+// If ServeFallbackSecret is true, FallbackSecretName is specified and there is a secret named
+// FallbackSecretName, return the fall back secret.
 func (sf *SecretFetcher) FindIngressGatewaySecret(key string) (secret model.SecretItem, ok bool) {
+	log.Debugf("FindIngressGatewaySecret: searching for secret %s", key)
+	log.Debugf("use fallback secret %v, fallback secret name %s", sf.ServeFallbackSecret, sf.FallbackSecretName)
 	val, exist := sf.secrets.Load(key)
+	log.Debugf("load secret %s from secret fetcher: %v", key, exist)
 	if !exist {
+		if sf.ServeFallbackSecret {
+			log.Debugf("Cannot find secret %s, searching for fallback secret %s", key, sf.FallbackSecretName)
```

Comment:
I think this should be warning level. It is always bad to fallback since that indicates that the secret does not exist -- they will want to take action on this. ideally a metric as well, but that can be followup

Reply:
Changed to Warnf, and added a TODO